pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -104,8 +121,26 @@ def get_batch_log_metrics(self, *args, **kwargs):
     def run_epoch_func(self, results, opt_metric, func_name, *args, **kwargs) -> None:
         if not isinstance(opt_metric, Result):
             raise Exception("The provided opt_metric should be a Result Object. Something is wrong")
+
         func = getattr(opt_metric, func_name)
         metrics_to_log = func(*args, add_dataloader_idx=self.has_several_dataloaders, **kwargs)
+        if (
+            torch.distributed.is_available() and torch.distributed.is_initialized()
+            and self._all_gather_fn.__self__.trainer.world_size > 1
```

Comment:
is this really the easiest way to check the world size?

Reply:
I tried, but I had some issues with windows using get_world_size utils from torch. Any idea ?

Reply:
Let me try to find a better way.